### PR TITLE
Log all episodes added to trakt watchlist and fix 'e' not referenced

### PR DESCRIPTION
**Pull request by fernandog**
as discussed with @ratoaq2 for now we will keep broad catch until we use a notifiy manager and rewrite providers

@medariox @p0psicles 

```
SNATCHQUEUE-MANUALSNATCH-305288 :: Adding Stranger Things S01E05 to Trakt watchlist
```

### CODE REVIEW

**Comment by fernandog** [comment]
@p0psicles @ratoaq2 

snatch failed because of timeout!

```
2016-08-19 19:01:16 DEBUG    SNATCHQUEUE-MANUALSNATCH-247808 :: [f3ebd83] Adding Suits S06E06 to Trakt watchlist
2016-08-19 19:01:46 DEBUG    SNATCHQUEUE-MANUALSNATCH-247808 :: [f3ebd83] Traceback (most recent call last):
  File "/home/osmc/SickRage/sickbeard/search_queue.py", line 412, in run
    self.success = search.snatchEpisode(search_result)
  File "/home/osmc/SickRage/sickbeard/search.py", line 198, in snatchEpisode
    notifiers.trakt_notifier.update_watchlist(result.show, data_episode=data_episode, update="add")
  File "/home/osmc/SickRage/sickbeard/notifiers/trakt.py", line 186, in update_watchlist
    trakt_api.request(trakt_url, data, method='POST')
  File "/home/osmc/SickRage/lib/traktor/trakt.py", line 113, in request
    data=data, verify=self.ssl_verify)
  File "/home/osmc/SickRage/lib/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/osmc/SickRage/lib/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/home/osmc/SickRage/lib/requests/adapters.py", line 479, in send
    raise ReadTimeout(e, request=request)
ReadTimeout: HTTPSConnectionPool(host='api.trakt.tv', port=443): Read timed out. (read timeout=30)


```

File: sickbeard/search.py [hunk 1]
```diff
@@ -181,21 +183,25 @@ def snatchEpisode(result):  # pylint: disable=too-many-branches, too-many-statem
...
 
+    if trakt_data:
+        data_episode = notifiers.trakt_notifier.trakt_episode_data_generate(trakt_data)
+        if data_episode:
+            # TODO Remove this broad catch when Traktor handle exceptions
+            try:
+                notifiers.trakt_notifier.update_watchlist(result.show, data_episode=data_episode, update="add")
+            except Exception as e:
+                logger.log(u"Failed to watchlist episodes in Trakt. Error: {0}".format(e), logger.DEBUG)
+
```
Comment (fernandog) [inline sickbeard/search.py]: @ratoaq2 do we need to add this "TODO" to pytest whitelist?

File: sickbeard/search.py [hunk 2]
```diff
@@ -181,21 +183,25 @@ def snatchEpisode(result):  # pylint: disable=too-many-branches, too-many-statem
...
 
+    if trakt_data:
+        data_episode = notifiers.trakt_notifier.trakt_episode_data_generate(trakt_data)
+        if data_episode:
+            # TODO Remove this broad catch when Traktor handle exceptions
+            try:
+                notifiers.trakt_notifier.update_watchlist(result.show, data_episode=data_episode, update="add")
+            except Exception as e:
+                logger.log(u'Failed to watchlist episodes in Trakt. Error: {0}'.format(e), logger.DEBUG)
+
```
Comment (fernandog) [inline sickbeard/search.py]: @ratoaq2 do we need to add this "TODO" to pytest whitelist?

File: lib/traktor/exceptions.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from requests.exceptions import (ConnectionError, Timeout, TooManyRedirects)
+from requests.exceptions import (ConnectionError, ReadTimeout, Timeout, TooManyRedirects)
```
Comment (ratoaq2) [inline lib/traktor/exceptions.py]: I believe that ReadTimeout extends Timeout

Comment (fernandog) [inline lib/traktor/exceptions.py]: if so why I got traceback?

Comment (fernandog) [inline lib/traktor/exceptions.py]: @ratoaq2 Im lost:

http://docs.python-requests.org/en/master/_modules/requests/exceptions/
```
class Timeout(RequestException):
    """The request timed out.

    Catching this error will catch both
    :exc:`~requests.exceptions.ConnectTimeout` and
    :exc:`~requests.exceptions.ReadTimeout` errors.
    """
```

Comment (fernandog) [inline lib/traktor/exceptions.py]:
```python

>>> from requests.exceptions import (ReadTimeout, Timeout)
>>> try:
...     raise ReadTimeout
... except Timeout:
...     print "ReadTimeout catched!"
...
ReadTimeout catched!
```

Comment (fernandog) [inline lib/traktor/exceptions.py]: inside SickRage folder run `python`

```python
>>> import sys; sys.path.insert(1, 'lib')
>>> from requests.exceptions import ReadTimeout
>>> from traktor import TraktException
>>>
>>> try:
...     raise ReadTimeout
... except TraktException:
...     print "ReadTimeout catched!"
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
requests.exceptions.ReadTimeout
>>>

```

**Comment by fernandog** [comment]
@OmgImAlexis fixed you issue with wrong token

**Comment by fernandog** [comment]
@ratoaq2 im done here

**Pull request merged**